### ISSUE TITLE
Able to reset Allocations on a CreditNote?

### ISSUE THREAD

**Issue by thekeogh** [calcinai/xero-php#251]
Hi, There appears to be a problem with `Allocations` on `CreditNote`'s when there are already allocations to it, where it tries to apply all allocations again.

For example

```
$creditnote = $xero->loadByGUID('Accounting\\CreditNote', 'id-of-creditnote');
// Let's assume this credit note has a total of $100 and it has already allocated $40 to one invoice

$allocation = new Allocation($xero);
$allocation->setAppliedAmount(60.00);

$invoice = new Invoice($xero);
$invoice->setInvoiceID('id-of-invoice');
$allocation->setInvoice($invoice);

$creditnote->addAllocation($allocation);
// Now calling $creditnote->getAllocations() will return the $40 and the new $60 I just added

$creditnote->save();
// Will now error with: "Credit Note cannot be allocated as amount is greater than amount outstanding"
```

What the above is trying to do is allocation the $60 AND the $40 again, therefore erroring back at me.

I edited `XeroPHP\Models\Accounting\CreditNote@addAllocation()` to NOT push my new `Allocation` to the existing list and all worked fine.

Basically need a way to NOT load the existing allocations and just send over the new? Is there a way to do this right now? What workaround can I do in the meantime, cos this is a fatal issue for me, as I can't see a way around it easily?

Any help appreciated!

Thanks

**Comment by thekeogh** [comment]
Hmm or am I suppose to load a new CreditNote and setCreditNoteID() on that? I'll try that when able

**Comment by thekeogh** [comment]
Ok this kinda worked, allocated it but threw an error about not being able to alter the creditnote, which makes sense looking st the uri, as it's trying to update that. 

I need to use the /Allocations endpoint, so effectively saving the allocation and not the creditnote, how do I do this?

Basically need a example of how to use the Allocations endpoint, not adding Allocations during credit note add/update

**Comment by calcinai** [comment]
I think this may have [come up in the past](#224).  I don't think you'd necessarily been going about it wrong, as it shouldn't matter how you get that `CreditNote` - it should work either way.

Thanks for all the debugging notes.  I think the core issue here is that we don't get any sort of ID for the allocation, so they're not really possible to track.  Have you got any ideas on this?

Otherwise, I'll contact the guys at Xero and see if they'll expose their ID to solve this issue.

**Comment by rodjsta** [comment]
Hey, @calcinai did you end up getting any further with this annoying issue?

**Comment by calcinai** [comment]
@rodjsta Unfortunately not. Have you tried contacting Xero recently?
